pkg/image/registry/imagestreamimage/rest.go

```diff
@@ -54,7 +46,7 @@ func ParseNameAndID(input string) (name string, id string, err error) {
 // Get retrieves an image by ID that has previously been tagged into an image stream.
 // `id` is of the form <repo name>@<image id>.
 func (r *REST) Get(ctx kapi.Context, id string) (runtime.Object, error) {
-	name, imageID, err := ParseNameAndID(id)
+	name, imageID, err := parseNameAndID(id)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
There's one more candidate for `api.JoinImageStreamTag(name, tag)` in line 88 of this file, where ISI is constructed.